boa_engine/fuzz/fuzz_targets/common.rs

```diff
@@ -0,0 +1,71 @@
+use boa_engine::context::Context;
+use boa_engine::syntax::ast::visitor::{VisitWith, VisitorMut};
+use boa_engine::syntax::ast::{Expression, StatementList};
+use boa_interner::Sym;
+use libfuzzer_sys::arbitrary;
+use libfuzzer_sys::arbitrary::{Arbitrary, Unstructured};
+use std::fmt::{Debug, Formatter};
+use std::ops::ControlFlow;
+
+pub struct FuzzData {
+    pub context: Context,
+    pub ast: StatementList,
+}
+
+impl<'a> Arbitrary<'a> for FuzzData {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        let mut context = Context::default();
+        let mut syms_available = Vec::with_capacity(8);
+        for c in 'a'..='h' {
+            syms_available.push(context.interner_mut().get_or_intern(&*String::from(c)));
+        }
```

Comment:
Any reason to only have these symbols? Also, I see a `TODO` below requesting arbitrary string literals.

Reply:
Yes -- this creates a fixed pool of symbols to use in the AST. The AST, when generated, just throws random symbol indices in; we need to fix them to well-known (non-keyword) items in the symbol interner. Additionally, we don't want to generate them dynamically from the fuzz data because this can introduce "noise" into our sample (basically, bytes used previously to make symbols now form part of the AST or vice versa, unaligning the AST-generating bytes). This is also the reason for the TODO -- string literals from the arbitrary data could introduce quite a bit of undesirable noise, but not doing so means we don't test string parsing/interning. I decided not to for this PR because it could introduce so much noise that it would render this unusable.